inst/shiny/server.R

```diff
@@ -218,6 +244,58 @@ shinyServer(function(input, output, session) {
       updateAllPdataInputs()
     }
   })
+ 
+  #Create UI for filtering samples based on annotations
+  output$FilterSamples <- renderUI({
+    annot = ''
+    if (!is.null(vals$counts)){
+      annot = colnames(pData(vals$counts))
+    }
+    selectInput("filteredSample", "Select Annotation:", c("none", annot))
+  })
+  observeEvent(input$filteredSample, {
+    output$filterSampleOptions <- renderUI({
+      if (input$filteredSample != "none")({
+        L = vector("list", 2)
+        L[[1]] = list(checkboxGroupInput("filterSampleChoices", 
+                                         label = "Select samples to keep",
+                                         choices = unique(pData(vals$counts)[,input$filteredSample])))
+        L[[2]] = list(actionButton("runFilterSample", "Filter"))
+        return(L)
+      }) else {
+        L = list()
+      }
+    })
+  })
+  
+  #Filter the selected samples
+  observeEvent(input$runFilterSample, {
+    filter = pData(vals$counts)[,input$filteredSample] %in% input$filterSampleChoices
+    vals$counts = vals$counts[,filter]
+  })
+   
+  #Create UI for filtering genes based on feature annotations
+  observeEvent(input$filteredFeature, {
+    output$filterFeatureOptions <- renderUI({
+      if (input$filteredFeature != "none")({
+        L = vector("list", 2)
+        L[[1]] = list(checkboxGroupInput("filterFeatureChoices", 
+                                         label = "Select features to keep",
+                                         choices = unique(fData(vals$counts)[,input$filteredFeature])))
```

Comment:
I'm still hitting an issue with this, when I load a dataset that has a feature annotation that has many possible features to filter. For example:

```{r}
library(singleCellTK)
library(scater)
data("maits_SCESet")
maits_SCESet <- createSCESet(countfile = maits_SCESet$counts,
                             annotfile = maits_SCESet$annot,
                             featurefile = maits_SCESet$features,
                             inputdataframes = TRUE)
singleCellTK(maits_SCESet)
```
With this dataset, select "primerid" in the "Select Feature" field. On my computer that freezes the toolkit for >30 seconds and I don't think this would be useful for the user, because they would have to search through a 16,000 item list to find the things they want to keep. Is there a way we can remove the annotations with many possible feature annotations to filter on?